### PR TITLE
add events search apis

### PR DESCRIPTION
**Pull request by junotx**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Add apis to search events

### CODE REVIEW

File: pkg/models/tenant/tenant.go [hunk 1]
```diff
@@ -200,6 +210,50 @@ func (t *tenantOperator) ListNamespaces(user user.Info, workspace string, queryP
...
+	}
+	decision, _, err := t.authorizer.Authorize(listEvts)
+	if err != nil {
+		klog.Error(err)
+		return nil, err
+	}
+
+	var workspaces []*tenantv1alpha1.Workspace
+	lw, err := t.ListWorkspaces(user, query.New())
+	if err != nil {
```
Comment (wansir) [inline pkg/models/tenant/tenant.go:230]: Need to filter out the workspace that current user have the right to list events.

File: pkg/models/tenant/tenant.go [hunk 2]
```diff
@@ -200,6 +210,50 @@ func (t *tenantOperator) ListNamespaces(user user.Info, workspace string, queryP
...
+		ResourceRequest: true,
+	}
+	decision, _, err := t.authorizer.Authorize(listEvts)
+	if err != nil {
+		klog.Error(err)
+		return nil, err
+	}
+
+	var workspaces []*tenantv1alpha1.Workspace
+	lw, err := t.ListWorkspaces(user, query.New())
```
Comment (wansir) [inline pkg/models/tenant/tenant.go:229]: Need to filter out the workspace that current user have the right to list events.

Comment (junotx) [reply pkg/models/tenant/tenant.go:229]: Ok, let me add that

Comment (benjaminhuo) [reply pkg/models/tenant/tenant.go:229]: Looks like the filtering you mention is done in the IntersectedNamespaces function @wansir

Comment (wansir) [reply pkg/models/tenant/tenant.go:229]: ListWorkspaces only returns workspaces of a specific user, but the user may not have permission to list events in all the namespaces, even if they can view namespaces. Filtering logic is necessary.

Comment (benjaminhuo) [reply pkg/models/tenant/tenant.go]: I see.

**Comment by benjaminhuo** [comment]
@junotx , there are conflicts that must be resolved

File: pkg/simple/client/events/interface.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package events
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"time"
+)
+
+type Client interface {
+	SearchEvents(filter *Filter, from, size int64, sort string) (*Events, error)
```
Comment (zryfish) [inline pkg/simple/client/events/interface.go:9]: How to do unit testing without connecting to a real elasticsearch service? Please add more test code to this interface, you can refer to LDAP interface which implements a fake client that can be used to do simple unit testing. This is what we try to achieve, running standalone ks-apiserver or ks-controller-manager without all dependency.

Comment (junotx) [reply pkg/simple/client/events/interface.go:9]: Now it is added

**Pull request merged**